### PR TITLE
dwallet-network/eth-light-client: added cli command `init-eth-state`

### PR DESCRIPTION
**Pull request by yuvalelmo**

### CODE REVIEW

File: crates/sui/src/ethereum_client_commands.rs [hunk 1]
```diff
@@ -140,4 +140,61 @@ pub(crate) async fn create_eth_dwallet(
         context,
         Call
     ))
-}
\ No newline at end of file
+}
+
+//todo(yuval): in future, we should load also the sync committee from binary data file.
+/// Updates the current Ethereum state and creates a LatestEthState object
```
Comment (yuvalelmo) [inline crates/sui/src/ethereum_client_commands.rs:146]:
```suggestion
/// Initializes a shared LatestEthereumState object in the dWallet network
/// with the given checkpoint.
/// This function should only be called once to initialize the Ethereum state.
/// After the state is initialized, the Ethereum state object ID is saved in the configuration,
/// and the state is updated whenever a new state is successfully verified.
// todo(yuval): in future, we should load also the sync committee from binary data file.```

File: crates/sui-sdk/src/sui_client_config.rs [hunk 2]
```diff
@@ -61,6 +61,34 @@ impl SuiClientConfig {
         })
     }
 
+    pub fn get_active_env_mut(&mut self) -> Result<&mut SuiEnv, anyhow::Error> {
+        let active_env = self.active_env.clone(); // Clone active_env to avoid immutable borrow
```
Comment (zeevmoney) [inline crates/sui-sdk/src/sui_client_config.rs:65]:
```suggestion
        let active_env = self.active_env.clone();
```

File: crates/sui-sdk/src/sui_client_config.rs [hunk 3]
```diff
@@ -61,6 +61,34 @@ impl SuiClientConfig {
...
+
+    pub fn get_env_mut(&mut self, alias: &Option<String>) -> Option<&mut SuiEnv> {
+        if let Some(alias) = alias {
+            self.envs.iter_mut().find(|env| &env.alias == alias)
+        } else {
+            self.envs.first_mut()
+        }
+    }
+
+    pub fn update_ethereum_state_object_id(&mut self, object_id: ObjectID) {
```
Comment (zeevmoney) [inline crates/sui-sdk/src/sui_client_config.rs:83]: return result, don't use `except()`, return an error if needed.

File: crates/sui/src/ethereum_client_commands.rs [hunk 4]
```diff
@@ -83,6 +85,9 @@ pub enum EthClientCommands {
         /// A Trusted checkpoint to initialize the state.
         #[clap(long)]
         checkpoint: String,
+        /// The corresponding Ethereum network
```
Comment (zeevmoney) [inline crates/sui/src/ethereum_client_commands.rs:88]:
```suggestion
        /// The corresponding Ethereum network.
```

File: crates/sui/src/ethereum_client_commands.rs [hunk 5]
```diff
@@ -149,3 +154,78 @@ pub(crate) async fn create_eth_dwallet(
         Call
     ))
 }
+
+/// Initializes a shared LatestEthereumState object in the dWallet network
+/// with the given checkpoint.
+/// This function should only be called once to initialize the Ethereum state.
+/// After the state is initialized, the Ethereum state object ID is saved in the configuration,
+/// and the state is updated whenever a new state is successfully verified.
+// todo(yuval): in future, we should load also the sync committee from binary data file.```
```
Comment (zeevmoney) [inline crates/sui/src/ethereum_client_commands.rs:163]:
```suggestion
```
open an issue.

File: crates/sui/src/ethereum_client_commands.rs [hunk 6]
```diff
@@ -149,3 +154,78 @@ pub(crate) async fn create_eth_dwallet(
...
+    );
+
+    let SuiClientCommandResult::Call(state) = latest_state else {
+        return Err(anyhow!("Can't get response."));
+    };
+
+    let latest_state_object_id = state
+        .object_changes
+        .clone()
+        .unwrap()
```
Comment (zeevmoney) [inline crates/sui/src/ethereum_client_commands.rs:205]: remove all `unwrap()`s

**Review by zeevmoney — CHANGES_REQUESTED** [review]
see comments

**Review by ycscaly — APPROVED** [review]

File: crates/sui-sdk/src/sui_client_config.rs [hunk 7]
```diff
@@ -56,6 +56,44 @@ impl SuiClientConfig {
...
+    pub fn get_active_env_mut(&mut self) -> Result<&mut SuiEnv, anyhow::Error> {
+        let active_env = self.active_env.clone();
+        match self.get_env_mut(&active_env) {
+            None => Err(anyhow!(
+                "Environment configuration not found for env [{}]",
+                active_env.as_deref().unwrap_or("None")
+            )),
+            Some(env) => Ok(env),
+        }
+    }
```
Comment (zeevmoney) [inline crates/sui-sdk/src/sui_client_config.rs:69]:
```suggestion
    fn get_active_env_mut(&mut self) -> Result<&mut SuiEnv, anyhow::Error> {
        self.get_env_mut(&self.active_env).ok_or_else(|| {
            anyhow!(
                "Environment configuration wasn't found for the environment: [{}]",
                self.active_env.as_deref().unwrap_or("None")
            )
        })
    }
```

File: crates/sui-sdk/src/sui_client_config.rs [hunk 8]
```diff
@@ -56,6 +56,44 @@ impl SuiClientConfig {
...
+
+    pub fn get_env_mut(&mut self, alias: &Option<String>) -> Option<&mut SuiEnv> {
+        if let Some(alias) = alias {
+            self.envs.iter_mut().find(|env| &env.alias == alias)
+        } else {
+            self.envs.first_mut()
+        }
+    }
```
Comment (zeevmoney) [inline crates/sui-sdk/src/sui_client_config.rs:77]:
```suggestion
    fn get_env_mut(&mut self, alias: &Option<String>) -> Option<&mut SuiEnv> {
        alias
            .as_ref()
            .and_then(|alias| self.envs.iter_mut().find(|env| &env.alias == alias))
            .or_else(|| self.envs.first_mut())
    }
```

File: crates/sui-sdk/src/sui_client_config.rs [hunk 9]
```diff
@@ -56,6 +56,44 @@ impl SuiClientConfig {
...
+            } else {
+                Err(anyhow!(
+                    "no Ethereum State object ID found for active environment `{}`",
+                    env.alias
+                ))
+            }
+        } else {
+            Err(anyhow!("no active environment found"))
+        }
+    }
```
Comment (zeevmoney) [inline crates/sui-sdk/src/sui_client_config.rs:96]:
```suggestion
    pub fn update_ethereum_state_object_id(
        &mut self,
        object_id: ObjectID,
    ) -> Result<(), anyhow::Error> {
        let env = self.get_active_env_mut()?;
        env.eth_client_settings
            .as_mut()
            .map(|config| config.state_object_id = Some(object_id))
            .ok_or_else(|| {
                anyhow!(
                    "no Ethereum State object ID found for active environment: [{}]",
                    env.alias
                )
            })
    }
```

**Review by zeevmoney — APPROVED** [review]
fix suggestions and merge.

**Pull request merged**